src/main/java/liquibase/ext/mongodb/statement/ListCollectionNamesStatement.java

```diff
@@ -0,0 +1,92 @@
+package liquibase.ext.mongodb.statement;
+
+/*-
+ * #%L
+ * Liquibase MongoDB Extension
+ * %%
+ * Copyright (C) 2021 Mastercard
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+import liquibase.ext.mongodb.database.MongoLiquibaseDatabase;
+import liquibase.nosql.statement.NoSqlQueryForListStatement;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import org.bson.Document;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static liquibase.ext.mongodb.statement.BsonUtils.toCommand;
+
+/**
+ * Gets a list of collection names via the database runCommand method
+ * For a list of supported options see the reference page:
+ *
+ * @see <a href="https://docs.mongodb.com/manual/reference/command/listCollections/">listCollections</a>
+ */
+@Getter
+@EqualsAndHashCode(callSuper = true)
+public class ListCollectionNamesStatement extends AbstractRunCommandStatement
+        implements NoSqlQueryForListStatement<MongoLiquibaseDatabase, String> {
+
+    static final String RUN_COMMAND_NAME = "listCollections";
+    public static final String FILTER = "filter";
+    public static final String CURSOR = "cursor";
+    public static final String FIRST_BATCH = "firstBatch";
+    public static final String NAME = "name";
+    public static final String AUTHORIZED_COLLECTIONS = "authorizedCollections";
+    public static final String NAME_ONLY = "nameOnly";
```

Comment:
My personal style would not to extract constants for these unless there's reuse as it detracts from readability in-situ. There's no performance penalty since the compiler will optimise the String literals anyway. 
Could these be `private static final...` too?

Noted that this is the style you would like in the project though, so will follow it moving forward.

Reply:
Yes, that's style and some of our Sonar rules related to magic values. I prefer the fields used by mongo driver to be as constants, even for the current set NAME is reused. For example Cosmos java driver is exposing constants for their fields which I'm using in the liquibase-cosmosdb extension. Related to whether the constants have to be private, I usually  don't like hiding the things that do not worth hiding, if they are potentially to be used in other Statements.
Hopefully this style aspects won't irritate you and you will continue contributing :)